specification/keyvault/Security.KeyVault.SecurityDomain/routes.tsp

```diff
@@ -0,0 +1,137 @@
+  @route("/securitydomain/download/pending")
+  @get
+  downloadPending is KeyVaultOperation<{}, SecurityDomainOperationStatus>;
+
+  /**
+   * Retrieves the Security Domain from the managed HSM. Calling this endpoint can be used to activate a provisioned managed HSM resource.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation necessary for Key Vault"
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "@operationId necessary to preserve original operationId"
+  @OpenAPI.operationId("HSMSecurityDomain_Download")
+  @pollingOperation(HsmSecurityDomain.downloadPending)
+  @useFinalStateVia("azure-async-operation")
+  @route("/securitydomain/download")
+  @post
+  download is KeyVaultOperation<
+    {
+      /**
+       * The Security Domain download operation requires customer to provide N certificates (minimum 3 and maximum 10) containing a public key in JWK format.
+       */
+      #suppress "@azure-tools/typespec-azure-core/casing-style" "Preserving original casing"
+      @body
+      CertificateInfoObject: CertificateInfoObject;
+    },
+    {
+      @statusCode _: 202;
+
+      /**
+       * The URI to poll for completion status.
+       */
+      @pollingLocation
+      @header("Azure-AsyncOperation")
+      azureAsyncOperation?: string;
+
+      /**
+       * The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.
+       */
+      @header("Retry-After") retryAfter?: int32;
+
+      /**
+       * The response contains the Security Domain that is being confirmed.
+       */
+      @body response: SecurityDomainObject;
+    }
+  >;
+
+  /**
+   * Retrieve Security Domain transfer key
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation necessary for Key Vault"
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "@operationId necessary to preserve original operationId"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-names" "Operation name is already established"
+  @OpenAPI.operationId("HSMSecurityDomain_TransferKey")
+  @route("/securitydomain/upload")
+  @get
+  transferKey is KeyVaultOperation<{}, TransferKey>;
+
+  /**
+   * Restore the provided Security Domain.
+   */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation necessary for Key Vault"
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "@operationId necessary to preserve original operationId"
+  @OpenAPI.operationId("HSMSecurityDomain_Upload")
+  @pollingOperation(HsmSecurityDomain.uploadPending)
+  @finalOperation(HsmSecurityDomain.uploadPending)
+  @useFinalStateVia("azure-async-operation")
+  @route("/securitydomain/upload")
+  @post
+  upload is KeyVaultOperation<
+    {
+      /**
+       * The Security Domain to be restored.
+       */
+      #suppress "@azure-tools/typespec-azure-core/casing-style" "Preserving original casing"
```

Comment:
The security_domain is not a name used on the wire, correct? If so, the only reason to keep casing would be generated code from the generated openapi2/swagger document. In other words, Preserving this name should not matter.

Reply:
True, this isn't actually sent over the wire. I can update the casing here and elsewhere